snooty/test_postprocess.py

```diff
@@ -143,10 +143,37 @@ def test_validate_ref_targets(backend: Backend) -> None:
         <literal><text>mongod</text></literal>
         </ref_role>""",
     )
+    # Assert that refs beginning with number work
+    paragraph = ast.children[5]
+    assert isinstance(paragraph, n.Parent)
+    ref_role = paragraph.children[2]
+    check_ast_testing_string(
+        ref_role,
+        """<ref_role
+        domain="std"
+        name="label"
+        target="100_stacked_example"
+        fileid="index">
+        <text>100 Stacked Examples</text>
+        </ref_role>""",
+    )
 
+    paragraph = ast.children[6]
+    assert isinstance(paragraph, n.Parent)
+    ref_role = paragraph.children[2]
+    check_ast_testing_string(
+        ref_role,
+        """<ref_role
+        domain="std"
+        name="label"
+        target="z100_stacked_example"
+        fileid="index">
+        <text>Z100 Stacked Examples</text>
+        </ref_role>""",
+    )
     # Check that undeclared targets raise an error
     diagnostics = backend.diagnostics[page_id]
-    assert len(diagnostics) == 1
+    assert len(diagnostics) == 2
```

Comment:
This test might be fragile: it depends on the order of the diagnostics staying the same, which might not be a guarantee we want to make.

better to test that diagnostics *contains* the right diagnostic types